twisted/python/threadpool.py

```diff
@@ -209,6 +210,25 @@ def stop(self):
             thread.join()
 
 
+    @defer.inlineCallbacks
+    def stopAndWait(self):
+        """
+        Post stop request to all working threads and wait for them to finish.
+
+        Returns a deferred which fires when all workers are terminated.
+        """
+        self.joined = True
+
+        while self.workers:
+            self.q.put(WorkerStop)
+            self.workers -= 1
```

Comment:
do we need to decrease the counter here? I was expecting that WorkerStop will do that automatically.

but my root question is. What you/we want to do here?

Send stop command for all workers ?


maybe we can have

```
for _ in xrange(len(self.workers)):
    self.q.put(WorkerStop)
```